### DISCUSSION TITLE
How do I delete globals within a type definition file?

### DISCUSSION THREAD

**Discussion by roblox-ui** [JohnnyMorganz/luau-lsp#700] [Q&A]
Using type definition files I'm using this LSP to create autocomplete and typechecking for a game's scripting engine. They use `loadstring()` with `setfenv()`. Within a custom environment you can define your own globals, this is mostly fine because I can overwrite and add globals, but I can't remove them. Is there any way to remove them? If not, can I fork this extension to get my desired behavior?:

**Comment by JohnnyMorganz** [comment]
Hey, very sorry for the delay, I missed this.

What kind of globals do you want to delete? Are they builtin Luau globals, or globals from the Roblox definitions file? You can disable the roblox definitions file in the settings. I don't know if there is any way to remove builtins though without forking Luau
